### DISCUSSION TITLE
Variable name collision for getSearchResultsUsing method

### DISCUSSION THREAD

**Discussion by intrepidws** [filamentphp/filament#2007] [Ideas]
I am using the admin panel and have a form with a select field which makes use of the `getSearchResultsUsing()` method, [outlined here](https://filamentphp.com/docs/2.x/forms/fields#select). This works fine, but when I try to use more advanced where clauses (outlined in [Laravel docs here](https://laravel.com/docs/9.x/eloquent-relationships#chaining-orwhere-clauses-after-relationships)), I run into an issue.

Basically, if I use the code from Laravel (which uses the `$query` variable in a way that I think has become pretty common/standard in the community), then it results in a variable name collision since Filament appears to require that the variable provided to the `getSearchResultsUsing()` method be named `$query`.

Obviously, I could just change the name of the variable provided to the query builder, which is what I have done in the provided code below. However, given that Laravel itself has more or less standardized on using the `$query` variable in those scenarios - I wonder if it wouldn't make sense to allow the `getSearchResultsUsing()` method to use something like `$search` instead.

If the name of the variable must be one value only, then there is probably nothing to be done here since it would be a large breaking change. But if it is possible to provide flexibility in the name of the variable provided to the `getSearchResultsUsing()` method, then it might make sense to allow that and then change the docs to to reflect something like `$search`.

I know this probably sounds like a stupid idea, but it feels like a rough edge that could be smoothed out to be more consistent with the overall Laravel landscape.

Code that works:

```php
Forms\Components\Select::make('parent_registrant_id')
    ->label('Parent Registrant')
    ->searchable()
    ->getSearchResultsUsing(function (string $query) {
        return Registrant::query()
            ->where(function (Builder $query_builder) use ($query) {
                return $query_builder
                    ->where('full_name', 'like', "%{$query}%")
                    ->orWhere('badge_number', 'like', "%{$query}%");
            })
            ->orderBy('full_name')
            ->limit(50)
            ->pluck('full_name', 'id');
    })
    ->getOptionLabelUsing(function ($value): ?string {
        return Registrant::find($value)?->full_name;
    }),
```

Ideal scenario code:

```php
Forms\Components\Select::make('parent_registrant_id')
    ->label('Parent Registrant')
    ->searchable()
    ->getSearchResultsUsing(function (string $search) {
        return Registrant::query()
            ->where(function (Builder $query) use ($search) {
                return $query
                    ->where('full_name', 'like', "%{$search}%")
                    ->orWhere('badge_number', 'like', "%{$search}%");
            })
            ->orderBy('full_name')
            ->limit(50)
            ->pluck('full_name', 'id');
    })
    ->getOptionLabelUsing(function ($value): ?string {
        return Registrant::find($value)?->full_name;
    }),
```

**Comment by danharrin** [comment]
No breaking change, we just need to add a `search` array key here - https://github.com/laravel-filament/filament/blob/2.x/packages/forms/src/Components/Select.php#L148 😄 

...and update the docs.
